pennylane/ops/qubit/parametric_ops.py

```diff
@@ -729,6 +729,548 @@ def control_wires(self):
+                [zeros, ones, zeros, zeros],
+                [zeros, zeros, ones, zeros],
+                [zeros, zeros, zeros, ones],
+            ]
+
+            return qml.math.stack([stack_last(row) for row in matrix], axis=-2)
+
+        return qml.math.diag([exp_part, 1, 1, 1])
+
+    @staticmethod
+    def compute_eigvals(phi):  # pylint: disable=arguments-differ
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.CPhaseShift00.eigvals`
+
+
+        Args:
+            phi (tensor_like or float): phase shift
+
+        Returns:
+            tensor_like: eigenvalues
+
+        **Example**
+
+        >>> qml.CPhaseShift00.compute_eigvals(torch.tensor(0.5))
+        tensor([0.8776+0.4794j, 1.0000+0.0000j, 1.0000+0.0000j, 1.0000+0.0000j])
+        """
+        if qml.math.get_interface(phi) == "tensorflow":
+            phi = qml.math.cast_like(phi, 1j)
+
+        exp_part = qml.math.exp(1j * phi)
+        ones = qml.math.ones_like(exp_part)
+        return stack_last([exp_part, ones, ones, ones])
+
+    @staticmethod
+    def compute_decomposition(phi, wires):
+        r"""Representation of the operator as a product of other operators (static method). :
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+
+
+        .. seealso:: :meth:`~.CPhaseShift00.decomposition`.
+
+        Args:
+            phi (float): rotation angle :math:`\phi`
+            wires (Iterable, Wires): wires that the operator acts on
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> qml.CPhaseShift00.compute_decomposition(1.234, wires=(0,1))
+        [PauliX(wires=[0]),
+        PhaseShift(0.617, wires=[0]),
+        PhaseShift(0.617, wires=[1]),
+        CNOT(wires=[0, 1]),
+        PhaseShift(-0.617, wires=[1]),
+        CNOT(wires=[0, 1]),
+        PauliX(wires=[0])]
+
+        """
+        decomp_ops = [
+            qml.PauliX(wires[0]),
+            qml.PauliX(wires[1]),
+            qml.PhaseShift(phi / 2, wires=[wires[0]]),
+            qml.PhaseShift(phi / 2, wires=[wires[1]]),
+            qml.CNOT(wires=wires),
+            qml.PhaseShift(-phi / 2, wires=[wires[1]]),
+            qml.CNOT(wires=wires),
+            qml.PauliX(wires[1]),
+            qml.PauliX(wires[0]),
+        ]
+        return decomp_ops
+
+    def adjoint(self):
+        return CPhaseShift00(-self.data[0], wires=self.wires)
+
+    def pow(self, z):
+        return [CPhaseShift00(self.data[0] * z, wires=self.wires)]
+
+    @property
+    def control_values(self):
```

Comment:
What is the purpose of this method? (cc @albi3ro)

Reply:
Correct me if I'm wrong but I think it has to do with that this is a controlled phase shift with dependency on the first Qubit being $\lvert 0 \rangle$.

Reply:
Right! I'm just not sure if we definitely need this defined on these classes. :thinking:

Reply:
Let's leave this :100: